examples/ticks/date_index_formatter.py

```diff
@@ -18,19 +18,19 @@
 
 import matplotlib.cbook as cbook
 from matplotlib.dates import DateFormatter, DayLocator
+from matplotlib.layout_engine import ConstrainedLayoutEngine
 import matplotlib.lines as ml
 from matplotlib.ticker import Formatter
 
 # Load a numpy record array from yahoo csv data with fields date, open, high,
 # low, close, volume, adj_close from the mpl-data/sample_data directory. The
 # record array stores the date as an np.datetime64 with a day unit ('D') in
 # the date column (``r.date``).
-r = (cbook.get_sample_data('goog.npz', np_load=True)['price_data']
-     .view(np.recarray))
+r = cbook.get_sample_data('goog.npz', np_load=True)['price_data'].view(np.recarray)
 r = r[:9]  # get the first 9 days
 
 fig, (ax1, ax2) = plt.subplots(nrows=2, figsize=(6, 6),
-                               constrained_layout={'hspace': .15})
+                               layout=ConstrainedLayoutEngine(hspace=0.15))
```

Comment:
I don't mind this, but obscure imports are sometimes not nice.  Maybe:

```suggestion
                               layout='constrained')
fig.get_layout_engine.set(hspace=0.15)
```

Reply:
I don't feel strongly either way, so done.